File: Projetos/ACBrLib/Demos/C#/Shared/ACBrLib.Core/PIXCD/PSP.cs
using System.ComponentModel;

namespace ACBrLib.Core.PIXCD
{
    public enum PSP
    {
        [Description("Bradesco")]
        pspBradesco = 0,

        [Description("Itaú")]
        pspItau = 1,

        [Description("Banco do Brasil")]
        pspBancoBrasil = 2,

        [Description("Santander")]
        pspSantander = 3,

        [Description("Shipay")]
        pspShipay = 4,

        [Description("Sicredi")]
        pspSicredi = 5,

        [Description("Sicoob")]
        pspSicoob = 6,

        [Description("PagSeguro")]
        pspPagSeguro = 7,

        [Description("GerenciaNet")]
        pspGerenciaNet = 8,

        [Description("PixPDV")]
        pspPixPDV = 9,

        [Description("Inter")]
        pspInter = 10,

        [Description("Ailos")]
        pspAilos = 11,

        [Description("Matera")]
        pspMatera = 12,

        [Description("Cielo")]
        pspCielo = 13,

        [Description("Mercado Pago")]
        pspMercadoPago = 14,
    }
}

File: Projetos/ACBrLib/Demos/Java/ACBr Common/src/com/acbr/pixcd/PSP.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.acbr.pixcd;

import com.acbr.pixcd.*;
import java.util.HashMap;
import java.util.Map;

public enum PSP {
        pspBradesco (0),
        pspItau (1),
        pspBancoBrasil (2),
        pspSantander (3),
        pspShipay  (4),
        pspSicredi (5),
        pspSicoob (6),
        pspPagSeguro (7),
        pspGerenciaNet (8),
        pspPixPDV (9),
        pspInter (10),
        pspAilos (11),
        pspMatera (12),
        pspCielo (13),
        pspMercadoPago (14);
    
    private static final Map<Integer, PSP> map;
    private final int enumValue;
    
    static {
        map = new HashMap<>();
        for (PSP value : PSP.values()) {
            map.put(value.asInt(), value);
        }
    }
    
    public static PSP valueOf(int value){
        return map.get(value);
    }
    
    private PSP(int id){
        this.enumValue = id;
    }
    
    public int asInt() {
        return enumValue;
    }
}
